lib/origen/sub_blocks.rb

```diff
@@ -232,6 +232,15 @@ def sub_blocks(*args)
     end
     alias_method :children, :sub_blocks
 
+    # Returns a hash containing all sub block gruops of given sub-block
+    def sub_block_groups(*args)
+      if args.empty?
+        @sub_block_groups ||= {}.with_indifferent_access
+      else
+        sub_block_groups(*args)
+      end
+    end
```

Comment:
How does this method work when called with args?

Reply:
Good catch-- yes this seems like a typo. 

I had to fix the `:instance` option case in the `sub_block` method to call `sub_block_group` (singular) as it failed a test case (line 299), so I assumed that this would check this as well but now it seems recursive in nature.  Maybe we don't have a test case? 

Let me run it some more with correction.